runtime/src/bank.rs

```diff
@@ -1424,6 +1462,18 @@ impl Bank {
             })
     }
 
+    // Determine if the bank is currently in an upgrade epoch, where only votes are permitted
+    fn upgrade_epoch(&self) -> bool {
+        match self.operating_mode() {
+            #[cfg(test)]
+            OperatingMode::Development => self.epoch == 0xdead, // Value assumed by `test_upgrade_epoch()`
+            #[cfg(not(test))]
+            OperatingMode::Development => false,
+            OperatingMode::Preview => false,
+            OperatingMode::Stable => self.epoch == Epoch::max_value(),
```

Comment:
Rough process for upgrading mainnet-beta from 1.1 to 1.2 is then:
1. Pick an epoch that's 1-2 weeks away, add it in here and ship a new 1.1 and 1.2 release 
2. Write up a little document that describes the exact 1.1 and 1.2 versions to upgrade from/to, discuss what happens if the upgrade fails (hard fork restart from the last root in the upgrade epoch probably into 1.2 instead of rolling back to 1.1), where the coordination is occurring (Discord), who is coordinating the time of death (if it happens)
3. Circulate the document to all the validators, exchanges
4. Wait for the upgrade epoch, then execute the document